Sources/Vehicle/Extensions/HAEntityExtension.swift

```diff
@@ -0,0 +1,215 @@
+import Foundation
+import HAKit
+import PromiseKit
+import Shared
+import SwiftUI
+
+extension HAEntity {
+    public static func getIconForDomain(domain: String, size: CGSize) -> UIImage? {
+        do {
+            let tmpEntity = try HAEntity(entityId: "\(domain).ha_ios_placeholder",
+                                         domain: domain, state: "",
+                                         lastChanged: Date(),
+                                         lastUpdated: Date(),
+                                         attributes: [:],
+                                         context: HAResponseEvent.Context(id: "", userId: nil, parentId: nil))
+            return tmpEntity.getIcon(size: size)
+        }
+        catch {
+            return nil
+        }
+    }
+    
+    func onPress(for api: HomeAssistantAPI) -> Promise<Void>{
+        let domain = domain
+        var service: String
+        switch (domain) {
+            case "lock":
+                service = state == "unlocked" ? "lock" : "unlock"
+            case "cover":
+                service = state == "open" ? "close_cover" : "open_cover"
+            case "button","input_button":
+                service = "press";
+            case "scene":
+                service = "turn_on";
+            default:
+                service = state == "on" ? "turn_off" : "turn_on";
+        }
+        return api.CallService(domain: domain, service: service, serviceData: ["entity_id": entityId])
+    }
+    
+    func getIcon(size: CGSize = CGSize(width: 64, height: 64), darkColor: UIColor = UIColor.white) -> UIImage?{
+        let icon = attributes.icon ?? ""
+        
+        var image: MaterialDesignIcons = MaterialDesignIcons.bookmarkIcon
+        
+        if icon.starts(with: "mdi:") {
```

Comment:
use `.normalizingIconString` to do this icon name manipulation

Reply:
.normalizingIconString is inaccessible due to internal protection level. Can I make it public?

Reply:
I have moved the extension to Shared, so now it is accessible and used.